src/main/java/io/cdap/plugin/servicenow/source/ServiceNowInputFormat.java

```diff
@@ -102,25 +95,19 @@ public static List<ServiceNowTableInfo> fetchTableInfo(SourceQueryMode mode, Ser
     return tableInfos;
   }
 
-  private static ServiceNowTableInfo getTableMetaData(String tableName, ServiceNowConnectorConfig conf,
-                                                      SourceValueType valueType, String startDate, String endDate) {
+  private static ServiceNowTableInfo getTableMetaData(String tableName, ServiceNowConnectorConfig conf) {
     // Call API to fetch first record from the table
     ServiceNowTableAPIClientImpl restApi = new ServiceNowTableAPIClientImpl(conf);
 
-    ServiceNowTableDataResponse response = restApi.fetchTableSchema(tableName, valueType, startDate, endDate,
-                                                                    true);
-    if (response == null) {
-      return null;
+    Schema schema = null;
+    int recordCount = 0;
+    try {
+      schema = restApi.fetchTableSchema(tableName);
+      recordCount = restApi.getTableRecordCount(tableName);
+    } catch (OAuthProblemException | OAuthSystemException e) {
+      throw new RuntimeException(e);
```

Comment:
Since only `auth` exceptions are captured so please add some user friendly message for better understanding.
Ignore if you are already handling it somewhere.

Reply:
Added an exception message. 093d31e47a15f475e4d355bcbbfcb76d1c71e158